recipes/spark-install.rb

```diff
@@ -57,17 +57,54 @@
     to actual_install_dir
   end
 else
-  raise "Invalid Apache Spark installation mode: #{install_mode}. 'package' or 'tarball' required."
+  fail "Invalid Apache Spark installation mode: #{install_mode}. 'package' or 'tarball' required."
 end
 
 unless Chef::Config[:solo]
   # Bind the worker to the real network interface. We are assuming that /etc/hosts or DNS
   # is set up so that this name resolves correctly.
-  node.override['apache_spark']["standalone"]["worker_bind_ip"] =
-    "#{node["hostname"]}#{domain_suffix}"
+  node.override['apache_spark']['standalone']['worker_bind_ip'] =
+    "#{node['hostname']}#{domain_suffix}"
 end
 
-local_dirs = node['apache_spark']['standalone']['local_dirs']
+# Allow specifying an interface name such as 'eth0' to make this work in single-node test
+# environments without specifying an explicit IP address.
+%w(master_host master_bind_ip).each do |key|
+  %w(eth0 eth1).each do |iface_name|
+    if node['apache_spark']['standalone'][key] == iface_name
+      iface_ip_addr = node['network']['interfaces'][iface_name]['addresses'].keys[1]
+      Chef::Log.info("Setting Spark #{key} to #{iface_ip_addr} (resolved from #{iface_name})")
+      node.override['apache_spark']['standalone'][key] = iface_ip_addr
+    end
+  end
+end
+
+if (node['csd-ec2-ephemeral'] || {})['mounts'] && node['csd-ec2-ephemeral']['mounts'].any?
+  local_dirs = node['csd-ec2-ephemeral']['mounts'].map do |mount|
+    ::File.join(mount[:mount_point], 'spark', 'local')
+  end
+  if local_dirs.empty?
+    local_dirs = (0..3).map { |i| "/mnt/ephemeral#{i}" }
+                 .select { |d| ::File.directory?(d) }
+                 .map { |d| ::File.join(d, 'spark', 'local') }
+    Chef::Log.warn(
+      'EC2 ephemeral mount list is empty. Setting Spark local directories based on existing ' \
+      "/mnt/ephemeral{0,1,2,3} directories: #{local_dirs}. This might not be correct. " \
+      "It is recommended that node['apache_spark']['local_dir'] is explicitly set instead."
+    )
+  else
+    Chef::Log.info('Setting Spark local directories automatically ' \
+                   "based on EC2 ephemeral devices: #{local_dirs}.")
+  end
+elsif node['apache_spark']['local_dir']
+  local_dirs = Array(node['apache_spark']['local_dir'])
+  local_dirs.each do |dir|
+    # Discourage using comma-separated strings in Chef attributes
+    fail "Spark local directory names cannot include a comma: #{dir}" if dir.include?(',')
+  end
+else
+  local_dirs = ['/var/local/spark']
+end
```

Comment:
This code was removed in an earlier commit. I am not sure how exactly it could come back in your pull request.